tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/HttpClient/SocketsHttpHandler/YarpForwarderHttpClientFactoryIntegration.cs

```diff
@@ -0,0 +1,83 @@
+    MinimumVersion = "1.1.0",
+    MaximumVersion = "2.*.*",
+    IntegrationName = IntegrationName)]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public class YarpForwarderHttpClientFactoryIntegration
+{
+    private const string IntegrationName = nameof(Configuration.IntegrationId.HttpMessageHandler);
+    private const IntegrationId IntegrationId = Configuration.IntegrationId.HttpMessageHandler;
+    private const IntegrationId SocketHandlerIntegrationId = Configuration.IntegrationId.HttpSocketsHandler;
+
+    /// <summary>
+    /// OnMethodBegin callback
+    /// </summary>
+    /// <typeparam name="TTarget">Type of the target</typeparam>
+    /// <typeparam name="TArg1">Type of the context</typeparam>
+    /// <param name="instance">Instance value, aka `this` of the instrumented method</param>
+    /// <param name="context">Instance of Yarp.ReverseProxy.Forwarder.ForwarderHttpClientContext</param>
+    /// <param name="handler">Instance of System.Net.Http.SocketsHttpHandler</param>
+    /// <returns>Calltarget state value</returns>
+    internal static CallTargetState OnMethodBegin<TTarget, TArg1>(TTarget instance, ref TArg1 context, ref System.Net.Http.SocketsHttpHandler handler)
+    {
+        // If our HttpClient and SocketsHttpHandler integrations are not enabled, do not modify the factory behavior
+        if (!Tracer.Instance.Settings.IsIntegrationEnabled(IntegrationId) || !Tracer.Instance.Settings.IsIntegrationEnabled(SocketHandlerIntegrationId))
+        {
+            return CallTargetState.GetDefault();
+        }
+
+        return new CallTargetState(scope: null, state: handler);
+    }
+
+    /// <summary>
+    /// OnMethodEnd callback
+    /// </summary>
+    /// <typeparam name="TTarget">Type of the target</typeparam>
+    /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+    /// <param name="exception">Exception instance in case the original code threw an exception.</param>
+    /// <param name="state">Calltarget state value</param>
+    /// <returns>A return value, in an async scenario will be T of Task of T</returns>
+    internal static CallTargetReturn OnMethodEnd<TTarget>(TTarget instance, Exception exception, in CallTargetState state)
+    {
+        // On net6.0+, the proxy will inject the current Activity into the request headers, using the propagator
+        // stored in the SocketsHttpHandler.ActivityHeadersPropagator field. This will overwrite the propagation
+        // headers that have already been set by our Datadog tracer's HttpClient instrumentation.
+        // To ensure that distributed tracing works properly, unset the ActivityHeadersPropagator so the
+        // trace context is not updated by Yarp.
+        if (state.State is System.Net.Http.SocketsHttpHandler handler)
+        {
+            handler.ActivityHeadersPropagator = null;
```

Comment:
Just wondering if it'd be better do set this to something like the below instead of `null`?
- https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.distributedcontextpropagator.createpassthroughpropagator?view=net-8.0#system-diagnostics-distributedcontextpropagator-createpassthroughpropagator
- or
- https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.distributedcontextpropagator.createnooutputpropagator?view=net-8.0#system-diagnostics-distributedcontextpropagator-createnooutputpropagator

Reply:
Thanks for pointing this out, I didn't know these existed.

It looks like `CreatePassThroughPropagator` won't work for us because it still injects the W3C trace context for the root (server) span, it just chooses this span instead of the new http client span.

However it does look like the `CreateNoOutputPropagator` will work because the inject operation becomes a no-op. This might be safer and better in the future if we decide later we want to make the no-op a little more dynamic. I'll test this locally and if it passes our tests I'll make this change!

Reply:
Made the change to replace the propagator with the one provided by `CreateNoOutputPropagator` in b37461ffb41511e344beb6884297ba37838de418

This looked good in my local testing so let's see the results of the CI testing!